types/node/events.d.ts

```diff
@@ -1,8 +1,16 @@
 declare module "events" {
     class internal extends NodeJS.EventEmitter { }
 
+    interface NodeEventTarget {
+        once(event: string | symbol, listener: (...args: any[]) => void): this;
+    }
+
+    interface DomEventTarget {
+        addEventListener(event: string, listener: (...args: any[]) => void, opts: { once: boolean }): any;
+    }
+
     namespace internal {
-        function once(emitter: EventEmitter, event: string | symbol): Promise<any[]>;
+        function once(emitter: NodeEventTarget | DomEventTarget, event: string | symbol): Promise<any[]>;
```

Comment:
Correct me if I'm wrong, but it seems to me that these changes do not meet the documentation.

Could you double check this file change with https://nodejs.org/dist/latest-v12.x/docs/api/events.html#events_events_once_emitter_name ?

Reply:
```
➜  node git:(feat/node-v12.11) node
Welcome to Node.js v12.11.1.
Type ".help" for more information.
> class MyEventTarget { addEventListener(name, fn, opts) { console.log('listening to', name, fn, opts); setTimeout(() =
> fn(123), 100); } } 
undefined
> events.once(new MyEventTarget(), 'test').then(data => console.log('received', data))
listening to test [Function] { once: true }
Promise { <pending> }
> received [ 123 ]
```

The documentation does point you to the DOM standard which I think I captured correctly here, you could argue that `opts: { once: boolean }` may not be required (although desired to prevent listener leak).

I intentionally did __not__ reference the DOM typings as this has caused name conflicts and issues in the past and would require DOM typings to be loaded for each node project.

Reply:
One could also argue that `symbol`s are technically not compatible with `addEventListener` so I might separate these into distinct functions